www/js/control/PopOpCode.jsx

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { Modal, StyleSheet } from 'react-native';
+import { Button, Text, IconButton, Dialog, useTheme } from 'react-native-paper';
+import { useTranslation } from "react-i18next";
+import QrCode from "../components/QrCode";
+
+const PopOpCode = ({visibilityValue, tokenURL, action, setVis}) => {
+    const { t } = useTranslation(); 
+    const { colors } = useTheme();
+    
+    return (
+        <Modal visible={visibilityValue} onDismiss={() => setVis(false)}
+        transparent={true}>
+            <Dialog visible={visibilityValue} 
+            onDismiss={() => setVis(false)}
+            style={styles.dialog(colors.elevation.level3)}>
+                <Dialog.Title>{t("general-settings.qrcode")}</Dialog.Title>
+                <Dialog.Content style={styles.content}>
+                    <QrCode value={tokenURL}></QrCode>
+                    <Text>{t("general-settings.qrcode-share-title")}</Text>
+                    <IconButton icon="share" onPress={() => action()} style={styles.button}/>
```

Comment:
I ended up setting `prepopulateQRMessage.message`, because I did not find a "text" option in that readME, but adding "message" and keeping "url" meant that the opCode now shows in the share sheet and can be saved to files. See [this comment](https://github.com/e-mission/e-mission-phone/pull/1009#issuecomment-1686686220) for a demo and more explanation. I also added the text to the modal!